### ISSUE TITLE
Prepend jdbc: to DatabaseMetaData.getURL result

### ISSUE THREAD

**Issue by fengyaocheng** [prestodb/presto#19312]
Current DatabaseMetaData.getURL implementation returns without jdbc:, but it should be included it as other databases do it.

new URI(url.substring("jdbc:".length()))

**Comment by misterjpapa** [comment]
@tdcmeehan - I've created a PR https://github.com/prestodb/presto/pull/23397 to address this issue. I'm a little concerned if there are callers of the method out there that have been appending the jdbc: prefix themselves and if the logic wasn't defensive code the URL could end up like **jdbc:**jdbc:presto://127.0.0.1:63442 eg: with an extra jdbc: prefix. Will a mention of the fix the release notes be enough to give the Presto userbase a heads-up and prevent regressions?

**Comment by tdcmeehan** [comment]
I agree that a release note should be sufficient.

**Comment by steveburnett** [comment]
Do the examples in https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/installation/jdbc.rst#connecting need to be updated based on the changes in #23397 ?

**Comment by misterjpapa** [comment]
@fengyaocheng - Granted I'm a new contributor to the Presto project, but in practice a bug/issue that has been fixed is usually closed only after the corresponding PR has been approved and merged. Although the PR is approved, it has not been merged yet, so it seems premature to close the bug as completed right now.

**Comment by misterjpapa** [comment]
> Do the examples in https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/installation/jdbc.rst#connecting need to be updated based on the changes in #23397 ?

@steveburnett  - no changes are needed in the examples you cited. Essentially this bug was pointing out that the PrestoDatabaseMeta.getURL method was returning an URL that did not match the published spec and examples you cited. I assume callers would have to have done some kind of post process to the resulting URL to match the spec/examples to get everything to work properly. Once this fix is merged and makes it to production, no post processing will be needed. As mentioned in my earlier comment, it will be important that folks check the release notes to understand the post processing is no longer needed OR hopefully their post processing logic was defensive enough to handle the fix gracefully.

**Comment by steveburnett** [comment]
@misterjpapa, your answer makes complete sense, thank you!
